packages/docs/docs/hosting/heroku.mdx

```diff
@@ -0,0 +1,84 @@
+# Heroku
+
+## Overview
+
+For EzBackend to be compatible with hosting on Heroku, there needs to be some additional configuration.
+
+The below configuration is for the recommended configuration on Heroku, `EzBackend` + `Postgres`. You will probably need to configure all the items below for EzBackend to work as expected
+
+## Port Binding
+
+### Fix
+
+In `app.start`
+
+```ts
+app.start({
+  backend: {
+    listen: {
+      address: '0.0.0.0',
+    },
+  },
+});
+```
+
+### Explanation
+
+By default, EzBackend will only listen on address `127.0.0.1`. Since Heroku assigns a random-ish port to each instance, it will throw the error
+
+```
+Error R10 (Boot timeout) -> Web process failed to bind to $PORT within 60 seconds of launch
+```
+
+To fix this, we need to listen on all addresses, or address `0.0.0.0`
+
+## Auth Callback Route
+
+### Fix
+
+In `.env`
+
+```bash
+PRODUCTION_URL=<Your heroku endpoint e.g. https:my-app.herokuapp.com>
+```
+
+### Explanation
+
+By default, EzBackend tries to obtain your url in order to properly set the callback URL for authentication callback routes. However, this results in EzBackend detecting the url without `https`
+
+We can however specify the `PRODUCTION_URL` manually in order to override the automated URL detection, which will allow authentication to properly work
+
+## Database Persistence
+
+### Fix
+
+In `Heroku`
+
+1. Create an instance of Heroku postgres
+2. Share the instance of Heroku postgres with your EzBackend deployment
+3. Share the URL of the Heroku postgres instance as an environment variable `DATABASE_URL`
+4. Turn on SSL for Heroku postgres
+
+In `app.start`
+
+```ts
+app.start({
+  backend: {
+    typeorm: {
+      type: 'postgres',
+      url: process.env.DATABASE_URL,
+      ssl: true,
```

Comment:
For me `ssl: true` gives me an error of `self signed certificate`, I had to add `    extra: {
      ssl: {
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        rejectUnauthorized: false,
      },
    },`
instead

Reply:
+1, i also had the same error of self signed certificate